src/compositor/output.c

```diff
@@ -913,6 +923,23 @@ wlc_output_set_sleep(wlc_handle output, bool sleep)
    wlc_output_set_sleep_ptr(convert_from_wlc_handle(output, "output"), sleep);
 }
 
+WLC_API void
+wlc_output_set_gamma(wlc_handle output, uint16_t size, uint16_t *r, uint16_t *g, uint16_t *b)
+{
+   wlc_output_set_gamma_ptr(convert_from_wlc_handle(output, "output"), size, r, g, b);
+}
+
+WLC_API uint32_t
+wlc_output_get_gamma_size(wlc_handle output)
+{
+   struct wlc_output *_output = convert_from_wlc_handle(output, "output");
+
+   if (_output->bsurface.api.get_gamma_size)
+      return _output->bsurface.api.get_gamma_size(&_output->bsurface);
```

Comment:
Won't that segfault if get_gamma_size is NULL for this particular backend (which in practice will happen for the x backend)